### PR TITLE
implement export registers feature

### PR DESCRIPTION
**Pull request by QuocTrung76**
As discussed in https://github.com/eclipse-cdt-cloud/vscode-peripheral-inspector/issues/28, I do a PR to support the export registers feature.
exportbutton

openSaveDialog

enterFilename

display

### CODE REVIEW

**Comment by thegecko** [comment]
Many file changes are just removing the last new line, could you revert those, please?

**Comment by thegecko** [comment]
Looks like it's stopped building :(

**Comment by QuocTrung76** [comment]
> Many file changes are just removing the last new line, could you revert those, please?

I finished rebasing after some pull requests were merged. Could you let me know about reviewing process?

File: package.json
Comment (thegecko) [inline package.json:158]: should exportAll also be hidden from the palette?

File: package.json
Comment (thegecko) [inline package.json:217]: Is there any reason we only allow export of specific nodes? Can we export any?

File: src/fileUtils.ts
Comment (thegecko) [inline src/fileUtils.ts:7]: New feature: Can we add a nice name based on the node selected?

**Review by thegecko — CHANGES_REQUESTED** [review]
Thanks for this PR, it works really well. Overall package size has increased from 4.8MB to 6.3MB which I'm happy to live with.

I've tested this in the desktop and browser, both work as expected 👍 

I've added some inline comments and questions if you wouldn't mind taking a look?

File: src/fileUtils.ts
Comment (jonahgraham) [reply src/fileUtils.ts:7]: Are you referring to `saveLabel` and `title` fields of [`SaveDialogOptions`](https://code.visualstudio.com/api/references/vscode-api#SaveDialogOptions)?

File: src/fileUtils.ts
Comment (jonahgraham) [inline src/fileUtils.ts:11]: IIUC when `showSaveDialog` returns `undefined` that means the user pressed cancel, so I don't think a warning message should be displayed in that case. @QuocTrung76   Have a quick test that when you press cancel that no warning message is shown

File: src/fileUtils.ts
Comment (thegecko) [reply src/fileUtils.ts:7]: No, the default file name suggested on save is `untitled.xml`. Is it possible for there to be a better default?

Comment (jonahgraham) [reply src/fileUtils.ts:7]: Thanks @thegecko

@QuocTrung76 that means `defaultUri` needs to be updated in SaveDialogOptions (I leave it up to you whether saveLabel or title also should be updated)

Comment (QuocTrung76) [reply src/fileUtils.ts:12]: Hi @thegecko , I set the default filename to 'registers.xml', is it fine or do you have any recommendations?
Hi @jonahgraham , the saveDialogOptions do not allow setting the default filename by options, I have modified a bit to put it.

File: package.json
Comment (QuocTrung76) [reply package.json:158]: yes, that's right, I have hidden it from the palette.

File: package.json
Comment (QuocTrung76) [reply package.json]: After reviewing the functionality, I found no technical reason to restrict the export to only peripheral.* nodes. I’ve updated the implementation to allow exporting any node in the peripheral-inspector.svd view

File: src/fileUtils.ts
Comment (QuocTrung76) [reply src/fileUtils.ts]: thank you, I removed the warning.

File: src/fileUtils.ts
Comment (thegecko) [reply src/fileUtils.ts:12]: `registers.xml` is better IMO. I was wondering if the name could be based on the node name you are exporting?

File: package.json
Comment (thegecko) [inline package.json:335]:
```suggestion
```

`view/item/context` is no longer used since this view is now rendered as a webview not a treeview.

File: package.json
Comment (thegecko) [reply package.json]: Great!

File: src/fileUtils.ts
Comment (thegecko) [reply src/fileUtils.ts:12]: I'm happy to leave this as just `registers.xml` for now 👍

**Comment by thegecko** [comment]
If you could just delete the `view/item/context` block, I think we are good to merge 👍

File: src/fileUtils.ts
Comment (thegecko) [reply src/fileUtils.ts]: confirmed 👍

File: package.json
Comment (QuocTrung76) [reply package.json]: I have deleted the view/item/context, thank you for reviewing.

**Comment by thegecko** [comment]
All works as expected and all feedback has been resolved, I'm merging it.

Thanks for your work @QuocTrung76

**Pull request merged**
